units/patch_intent.go

```diff
@@ -407,7 +408,14 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 	patchDoc.ProjectStorageMethod = ppStorageMethod
 
 	if err = patchDoc.Insert(); err != nil {
-		return errors.Wrapf(err, "inserting patch '%s'", patchDoc.Id.Hex())
+		// If this is GH merge queue and the patch gets tried again
```

Comment:
I don't think this kind of problem (of duplicate insertions due to the job re-dispatching) is limited to the GH merge queue, the job could try again for any patch type if it gets interrupted midway through running. Can we update the comment to say it could happen for any patch, but GH merge queue patches are of particular concern (for the reasons stated below)?

The rest of the comment about the GH merge queue is valuable context though!